### DISCUSSION TITLE
GTK 4 support priority

### DISCUSSION THREAD

**Discussion by baybal** [romgrk/node-gtk#298] [Ideas]
With GJS possibly not making it into the GTK 4 world, I think we can have a bit of GJS mindshare drifting here if GTK 4 becomes a priority.

GJS first rolled out GTK 4 support a year ago, but since, I believe, started to having hard time following GTK upstream for reasons not much dissimilar to node-gtk's. I'd say there is always a dozen complete showstopper bugs in https://gitlab.gnome.org/GNOME/gjs/-/issues

4.0's APIs are nowhere near close to stabilising, and it may fit node-gtk better as a younger project without too much existing code to care about.

I also have very few hopes 3.0 being supported as much as 2.0 was, as GTK is becoming a nearly completely RedHat driven project. With RedHat being a commercial enterprise, I doubt they will do any charity in form of developer hours.

**Comment by romgrk** [comment]
I favor GTK 4 support over GTK 3 support, but I currently don't see any major issue to using GTK 4, other than the existing bugs in node-gtk that aren't GTK4 specific. As always, developer hours are hard to come by. I've been switching jobs and don't have much time these days.
